gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1642,6 +1669,10 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 					}
 				}
 
+				if ftype == TRANS_REF_X {
```

Comment:
Add an explanatory comment?

Reply:
Added it.  While looking at this again, I also noticed a case that wasn't previously covered, the double reference -- something like `&(&value)` -- so I've fixed that here as well. https://github.com/gnolang/gno/pull/2731/commits/b33f0b0b284299b704ea7a4dade76c31e87d743a

The first fix was premature. I've corrected it here so that the error message is more correct. https://github.com/gnolang/gno/pull/2731/commits/71afb10ef07a506079180955b32e41c22627c3b5